internal/component/otelcol/connector/count/count.go

```diff
@@ -0,0 +1,105 @@
+package count
+
+import (
+	"github.com/grafana/agent/internal/component"
+	"github.com/grafana/agent/internal/component/otelcol"
+	"github.com/grafana/agent/internal/component/otelcol/connector"
+	"github.com/grafana/agent/internal/featuregate"
+	"github.com/grafana/river"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector"
+	otelcomponent "go.opentelemetry.io/collector/component"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:      "otelcol.connector.count",
+		Stability: featuregate.StabilityExperimental,
+		Args:      Arguments{},
+		Exports:   otelcol.ConsumerExports{},
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := countconnector.NewFactory()
+			return connector.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.connector.count component.
+type Arguments struct {
+	Spans      []MetricInfo `river:"span,block,optional"`
+	SpanEvents []MetricInfo `river:"spanevent,block,optional"`
+	Metrics    []MetricInfo `river:"metric,block,optional"`
+	DataPoints []MetricInfo `river:"datapoint,block,optional"`
+	Logs       []MetricInfo `river:"log,block,optional"`
```

Comment:
It would be good to write a test which uses all of these values at once. I wonder whether it would work. Issues such as #6329 might prevent the component from working.

Reply:
added the `Test_ComponentIO` test and it seems to work